lib/bookmarks_bulk_action.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class BookmarksBulkAction
+  def initialize(user, bookmark_ids, operation, options = {})
+    @user = user
+    @bookmark_ids = bookmark_ids
+    @operation = operation
+    @changed_ids = []
+    @options = options
+  end
+
+  def self.operations
+    @operations ||= %w[clear_reminder delete]
+  end
+
+  def perform!
+    unless BookmarksBulkAction.operations.include?(@operation[:type])
+      raise Discourse::InvalidParameters.new(:operation)
+    end
+    # careful these are private methods, we need send
+    send(@operation[:type])
+    @changed_ids.sort
+  end
+
+  private
+
+  def delete
+    @bookmark_ids.each do |b_id|
+      if guardian.can_delete?(b_id)
+        BookmarkManager.new(@user).destroy(b_id)
+        @changed_ids << b_id
+      end
+    end
+  end
+
+  def clear_reminder
+    bookmarks.each do |b|
+      if guardian.can_edit?(b)
+        BookmarkReminderNotificationHandler.new(b).clear_reminder
+        @changed_ids << b.id
+      else
+        raise Discourse::InvalidAccess.new
```

Comment:
This means that if you have one id of a bookmark that you can't edit, the operation will stop there and ignore the remaining bookmarks? We're not doing the same for deletes... it would be nice to have something consistent.

Reply:
When I was adding the tests, I noticed that the exception was being raised for delete action because of how [BookmarkManager](https://github.com/discourse/discourse/blob/2347ff707455e6da6447f12f3715a3bded408cc0/lib/bookmark_manager.rb#L130C1-L131C1) works. The BookmarkManager is raising the Invalid access error for delete operations. I wanted to do the same for clear reminders to keep it consistent. 
If you think it is better, I can remove the error, but I wanted to give you the full context of why I added it.

Reply:
Ah, that makes sense, we are doing the same for deletes. I think you're right to keep the consistency. Maybe we move the check inside of `BookmarkReminderNotificationHandler.new(b).clear_reminder` so that individual reminder clearances also get the access check?

Reply:
I started this update and noticed that it would affect places where I'm unsure of the impact.

To add this check inside ```BookmarkReminderNotificationHandler``` I need to pass the user as an argument when a new ```BookmarkReminderNotificationHandler``` is initialized. This would require updates in 3 files and updates in its correspondent tests.

I think it would be better to change this in a separate PR to keep the scope of this one smaller. It would also be easier to change and possibly revert if something goes wrong. 

Let me know your thoughts.

Reply:
Apologies for the delay, Amanda! 

> I think it would be better to change this in a separate PR to keep the scope of this one smaller. It would also be easier to change and possibly revert if something goes wrong.
> 

Yes, this makes sense to me!